tests/test/util/test_scheduling.cpp

```diff
@@ -93,4 +95,283 @@ TEST_CASE("Test converting point-to-point mappings to scheduling decisions",
     REQUIRE(actual.messageIds == expectedMessageIds);
     REQUIRE(actual.hosts == expectedHosts);
 }
+
+class SchedulingDecisionTestFixture : public SchedulerTestFixture
+{
+  public:
+    SchedulingDecisionTestFixture() { faabric::util::setMockMode(true); }
+
+    ~SchedulingDecisionTestFixture() { faabric::util::setMockMode(false); }
+
+  protected:
+    int appId = 123;
+    int groupId = 456;
+    std::string masterHost = faabric::util::getSystemConfig().endpointHost;
+
+    // Helper struct to configure one scheduling decision
+    struct SchedulingConfig
+    {
+        std::vector<std::string> hosts;
+        std::vector<int> slots;
+        int numReqs;
+        bool forceLocal;
+        SchedulingTopologyHint topologyHint;
+        std::vector<std::string> expectedHosts;
+    };
```

Comment:
I opted for the struct so that each test configuration is packed nicely and can be passed around easily. Plus, I find it helps tidying up the test set-up process.

I am slightly reluctant to remove it as I feel it helps with readability. Do you have any strong feelings regarding actually keeping it?